include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm_banded.hpp

```diff
@@ -0,0 +1,322 @@
+     *
+     * In the banded alignment the iteration of the inner columns is split into two phases. The first phase
+     * reuses the unbanded column computation and assumes that the banded score matrix always starts at the beginning
+     * of the matrix. In the second phase the special interface
+     * seqan3::detail::pairwise_alignment_algorithm_banded::compute_band_column is used to compute the banded column.
+     * The current implementation assumes that the first cell of the current score matrix column is always the first
+     * cell to compute in every column. The respective seqan3::detail::score_matrix_single_column is resized to the band
+     * size plus one additional field to cover the end of the band where the end of the column is not reached yet.
+     * This cell will never be written to but only read from, i.e. it represents minus infinity. This allows the
+     * algorithm to reuse the standard unbanded cell computation. The following figure depicts the referenced
+     * cell of the underlying score matrix (assuming seqan3::detail::score_matrix_single_column):
+     *
+     *       A G G T C A
+     *     0 1 2 3 4 5 6
+     *    |–|—|—|—|—|—|—|
+     *   0|0|0|0|0|0| | |
+     * A 1|1|1|1|1|1|0| |
+     * C 2|x|2|2|2|2|1|0|
+     * G 3| |x|3|3|3|2|1|
+     * T 4| | |x|4|4|3|2|
+     *
+     * The coordinate matrix represents the global matrix index and not the local band coordinate. Data structures that
+     * require the coordinate might need to map the global matrix coordinate to their local coordinate:
+     *
+     *             A     G     G     T     C     A
+     *       0     1     2     3     4     5     6
+     *    |–––––|–––––|–––––|–––––|–––––|–––––|–––––|
+     *   0|(0,0)|(0,1)|(0,2)|(0,3)|(0,4)|     |     |
+     * A 1|(1,0)|(1,1)|(1,2)|(1,3)|(1,4)|(1,5)|     |
+     * C 2|     |(2,1)|(2,2)|(2,3)|(2,4)|(2,5)|(2,6)|
+     * G 3|     |     |(3,2)|(3,3)|(3,4)|(3,5)|(3,6)|
+     * T 4|     |     |     |(4,3)|(4,4)|(4,5)|(4,6)|
+     *
+     * \throws seqan3::invalid_alignment_configuration if the band is not valid for the given sequences or
+     *         std::bad_alloc if too much memory is allocated.
+     */
+    template <std::ranges::forward_range sequence1_t, std::ranges::forward_range sequence2_t>
+    int32_t compute_matrix(sequence1_t && sequence1, sequence2_t && sequence2)
+    {
+        size_t sequence1_size = std::ranges::distance(sequence1);
+        size_t sequence2_size = std::ranges::distance(sequence2);
+
+        check_valid_band_configuration(sequence1_size, sequence2_size);
+
+        // ---------------------------------------------------------------------
+        // Initialisation phase: allocate memory and initialise first column.
+        // ---------------------------------------------------------------------
+
+        this->reset_optimum(); // Reset the tracker for the new alignment computation.
+
+        thread_local score_matrix_single_column<int32_t> local_score_matrix{};
```

Comment:
Don't you use the user provided `score_type`?

Reply:
Okay let's postpone this to an after PR fix.

Reply:
Ah okay you chose `score_type` = int, because it has the best performance, but shouldn't we make this an at least type? So if the user provides us `int64_t`, we should use `int64_t`, because he says us that the score can be BIG.

Reply:
It will be all resolved when the vectorised code is merged. Then the user can choose whatever he want :). Please note, that this is a step by step replacing the old code with the new more performant code. It will only pull in features step by step.